File: elixir/bob/bob.exs
defmodule Bob do
  def hey(input) do
    cond do
      silence?(input) -> "Fine. Be that way!"
      question?(input) -> "Sure."
      shouting?(input) -> "Whoa, chill out!"
      true -> "Whatever."
    end
  end

  def silence?(input) do
    String.strip(input) == ""
  end

  def question?(input) do
    String.last(input) == "?"
  end

  def shouting?(input) do
    String.upcase(input) == input && String.downcase(input) != input
  end
end

File: clojure/bob/src/bob.clj
(ns bob)
(use 'clojure.string)

(defn shouting?
  [phrase]
  (and
    (= phrase (upper-case phrase))
    (not= phrase (lower-case phrase))))

(defn question?
  [phrase]
  (ends-with? phrase "?"))

(defn response-for
  [phrase]
  (cond
    (blank? phrase) "Fine. Be that way!"
    (shouting? phrase) "Whoa, chill out!"
    (question? phrase) "Sure."
    :else "Whatever."))
